### ISSUE TITLE
Control F & Control B stops working with really really long line

### ISSUE THREAD

**Issue by lando2319** [XVimProject/XVim#950]
Create a line that is so long it takes up the entire screen (with line wrapping on). Once the line is long enough to take up the whole screen, the control F and control B functions stop working. 

Huge XVim fan.
Running Mac OS 10.11.4, XCode Version: 7.3

**Comment by hamsternik** [comment]
I update to Xcode 7.3.1, install XVim and i don't gain this problem.

@lando2319 and what about `Ctrl-B`, `Ctrl-D` and so others vim keymaps?

**Comment by lando2319** [comment]
Just upgraded to XCode 7.3.1 (and did git pull on XVim) the issue is still
present. It happens with: <c-f>, <c-d>, Interestingly <c-e> DOES work.

The bindings to move up all work (<c-b>, <c-u, <c-y)

Kind Regards

On Sun, May 8, 2016 at 4:55 PM Nikita Khomitsevich <notifications@github.com>
wrote:

> I update to Xcode 7.3.1, install XVim and i don't gain this problem.
>
> @lando2319 <https://github.com/lando2319> and what about Ctrl-B, Ctrl-D
> and so others vim keymaps?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/XVimProject/XVim/issues/950#issuecomment-217748627>
>
-- 
I make MVPs. Independent iOS Developer
Mike P Land - http://mikepland.com/

**Comment by hamsternik** [comment]
@lando2319 so..it's very strange, because all default command work correctly within my .xvimrc config. 

So, please, put your line of code for which def. vim commands don't work. I think, it'll help for resolve your problem.

**Comment by lando2319** [comment]
My xvimrc doesn't have anything related to those commands. This of course
only happens with really really long lines. I'm sure it's not working at
least for machine.

https://github.com/lando2319/dot_files_mpl/blob/master/.xvimrc

On Mon, May 9, 2016, 6:51 PM Nikita Khomitsevich <notifications@github.com>
wrote:

> @lando2319 <https://github.com/lando2319> so..it's very strange, because
> all default command work correctly within my .xvimrc config.
>
> So, please, put your line of code for which def. vim commands don't work.
> I think, it'll help for resolve your problem.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/XVimProject/XVim/issues/950#issuecomment-218022902>
>
-- 
I make MVPs. Independent iOS Developer
Mike P Land - http://mikepland.com/

**Comment by hamsternik** [comment]
@lando2319 yes, as i thought, the reason isn't in remapped keys. And what about a rather long code line, for which control F/B doesn't work, can you put it here? (cause i can't reproduce this bug)

**Comment by lando2319** [comment]
Sure thing copy and paste the contents of this gist into a playground (or
anywhere in xcode) and see if you can scroll past the long line using <c-f>
or <c-d>

https://gist.github.com/lando2319/87c09a24afe74b3e15d6d284cc5a0ea9


On Tue, May 10, 2016 at 8:03 AM Nikita Khomitsevich <
notifications@github.com> wrote:

> @lando2319 <https://github.com/lando2319> yes, as i thought, the reason
> isn't in remapped keys. And what about a rather long code line, for which
> control F/B doesn't work, can you put it here? (cause i can't reproduce
> this bug)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/XVimProject/XVim/issues/950#issuecomment-218150557>
>
-- 
I make MVPs. Independent iOS Developer
Mike P Land - http://mikepland.com/

**Comment by hamsternik** [comment]
@lando2319 so, the reason of non-scrolling behaviour for vim keymaps that is in your example long comment line filling up more lines in Xcode, than the `ctrl-u` can jump. For this, i can jump by using `ctrl-f`. if you create a so big line which will fill more than 74 lines in code as one whole line, `ctrl-f` won't work.

What is strange, i can move using `ctrl-e` easily.

**Comment by hamsternik** [comment]
@lando2319 and yes, all *scroll up* keymaps are working perfectly.
